subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -2307,6 +2310,47 @@ static int handle_request(struct coap_packet *request, struct lwm2m_message *msg
 	}
 #endif
 
+#if defined(CONFIG_LWM2M_GATEWAY_OBJ_SUPPORT)
```

Comment:
I'd rather see this code encapsulated into some helper function (likely implemented along the gateway object implementation) and only called here. I'm not in favor of bloating generic code with object-specific stuff.
For example something like `lwm2m_gw_request_handle()` which could return 0 if the message was consumed (was prefixed), 1 if the engine should process it as a regular message (unprefixed) or negative on error.